src/ui/public/query_bar/directive/query_popover.js

```diff
@@ -0,0 +1,128 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { uiModules } from 'ui/modules';
+import { documentationLinks } from '../../documentation_links/documentation_links';
+import {
+  EuiPopover,
+  EuiButtonEmpty,
+  EuiForm,
+  EuiFormRow,
+  EuiSwitch,
+  EuiLink,
+  EuiText,
+  EuiSpacer,
+} from '@elastic/eui';
+
+const luceneQuerySyntaxDocs = documentationLinks.query.luceneQuerySyntax;
+const kueryQuerySyntaxDocs = documentationLinks.query.kueryQuerySyntax;
+
+const module = uiModules.get('app/kibana', ['react']);
+module.directive('queryPopover', function (localStorage) {
+
+  return {
+    restrict: 'E',
+    scope: {
+      language: '<',
+      onSelectLanguage: '&',
+    },
+    link: function ($scope, $element) {
+      $scope.isPopoverOpen = false;
+
+      function togglePopover() {
+        $scope.$evalAsync(() => {
+          $scope.isPopoverOpen = !$scope.isPopoverOpen;
+        });
+      }
+
+      function closePopover() {
+        $scope.$evalAsync(() => {
+          $scope.isPopoverOpen = false;
+        });
+      }
+
+      function onSwitchChange() {
+        $scope.$evalAsync(() => {
+          const newLanguage = $scope.language === 'lucene' ? 'kuery' : 'lucene';
+          localStorage.set('kibana.userQueryLanguage', newLanguage);
+          $scope.onSelectLanguage({ $language: newLanguage });
+        });
+      }
+
+      function render() {
+        const button = (
+          <EuiButtonEmpty
+            size="xs"
+            flush="right"
+            onClick={togglePopover}
+          >
+            Options
+          </EuiButtonEmpty>
+        );
+
+        const popover = (
+          <EuiPopover
+            id="popover"
+            ownFocus
+            anchorPosition="downRight"
+            button={button}
+            isOpen={$scope.isPopoverOpen}
+            closePopover={closePopover}
+          >
+            <div style={{ width: '350px' }}>
+              <EuiText>
+                <p>
+                  Our experimental autocomplete and simple syntax features can help you create your queries. Just start
```

Comment:
Worth adding something about losing recent search results? I realize it's always been that way, but now we're making it easier to opt in to kuery and I wonder if it'll confuse users.

Reply:
You mean the search history in the autocomplete dropdown? There was some debate around this copy in Slack and this is what everyone settled on, so I'm a bit hesitant to change it for fear of starting another debate :) 

(you shouldn't lose your lucene history btw, not sure if that's what you meant, I assume you're just talking about how the histories for the two languages are separate, but the old lucene history should still be there if you switch back, let me know if that's not the case)

Reply:
Yeah, that's what I meant. You don't really lose it, it comes back when you switch back to lucene, but users probably don't realize that. I can see them turning it on and then being confused that all their recent searches disappear. Maybe something to add to the docs at least? I'm good with it, just thought I'd ask.